spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/AppDeployerReleaseManager.java

```diff
@@ -275,7 +277,17 @@ public Release delete(Release release) {
 		List<String> deploymentIds = appDeployerData.getDeploymentIds();
 		if (!deploymentIds.isEmpty()) {
 			for (String deploymentId : deploymentIds) {
-				appDeployer.undeploy(deploymentId);
+
+				// don't error trying trying to undeploy something
+				// which is not deployed
+				AppStatus appStatus = appDeployer.status(deploymentId);
+				if (appStatus.getState().equals(DeploymentState.deployed)) {
```

Comment:
In case of multiple instances requested for the app, the appStatus could be `partial` and we expect to `undeploy` the ones that are deployed in that case as well.

Reply:
This was copied from `DeleteStep` as is to check same case as otherwise call to undeploy would fail. Would you complain same thing in `DeleteStep` as well?